app/scripts/controllers/cluster.js

```diff
@@ -61,10 +61,16 @@ angular.module('cluster', ['stats', 'sql', 'common', 'nodeinfo'])
         getUpdatedNodeList($scope.nodeId);
         redirectToFirstNode();
 
-        $scope.$on('clusterState.refreshed', function () {
+        ClusterEventsHandler.register('clusterState_refreshed', 'ClusterController', function () {
           getUpdatedNodeList($scope.nodeId);
           redirectToFirstNode();
         });
+
+        $scope.$on('$destroy', function () {
+          //remove call back when controller is destroyed
+          ClusterEventsHandler.remove('clusterState_refreshed', 'ClusterController');
```

Comment:
optional: make a constant for the event,
e.g. `ClusterEventsHandler.STATE_REFRESHED` and `ClusterEventsHandler.CHECKS_REFRESHED`